app/routes/events/components/EventAttendeeStatistics.tsx

```diff
@@ -253,13 +259,36 @@ const Analytics = ({ eventId }: { eventId: ID }) => {
 
   const dispatch = useAppDispatch();
 
+  const memoizedData = useMemo(() => {
+    return async () => {
+      if (eventId) {
+        const response = await dispatch(fetchAnalytics(eventId));
+        let data = response.payload;
+
+        if (viewStartTime) {
+          data = data.filter(
+            (item) => new Date(item.date) >= new Date(viewStartTime as string)
+          );
+        }
+
+        if (viewEndTime) {
+          data = data.filter(
+            (item) => new Date(item.date) <= new Date(viewEndTime as string)
+          );
+        }
+        return data;
+      }
+    };
+  }, [eventId, viewStartTime, viewEndTime, dispatch]);
```

Comment:
The API returns complete data, regardless of what the user has set has `viewStartTime` and `viewEndTime`.

I recommend moving all of the filtering logic into your `useEffect` statement below, and removing these fields from the `useMemo`'s dependency array
```suggestion
  }, [eventId, dispatch]);
```